core/src/avm2/globals/flash/display/display_object.rs

```diff
@@ -21,7 +20,51 @@ use crate::vminterface::Instantiator;
 use crate::{avm2_stub_getter, avm2_stub_setter};
 use ruffle_render::filters::Filter;
 use std::str::FromStr;
-use swf::BlendMode;
+use swf::Twips;
+use swf::{BlendMode, Rectangle};
+
+pub fn display_object_allocator<'gc>(
+    class: ClassObject<'gc>,
+    activation: &mut Activation<'_, 'gc>,
+) -> Result<Object<'gc>, Error<'gc>> {
+    let class_name = class
+        .inner_class_definition()
+        .read()
+        .name()
+        .to_qualified_name(activation.context.gc_context);
```

Comment:
This is wrong because it prints the fully qualified class name: for example, in Ruffle, `new DisplayObject()` prints `ArgumentError: Error #2012: flash.display:DisplayObject$ class cannot be instantiated.`, while in Flash it prints `ArgumentError: Error #2012: DisplayObject$ class cannot be instantiated.`
(it's probably worth it adding a test that checeks the error message for `new DisplayObject()`.)

Reply:
Fixed